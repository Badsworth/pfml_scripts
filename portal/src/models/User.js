import BaseModel from "./BaseModel";

class User extends BaseModel {
  get defaults() {
    return {
      user_id: null,
      auth_id: null,
      email_address: null,
      first_name: null,
      middle_name: null,
      last_name: null,
      date_of_birth: null,
      has_state_id: null,
      state_id: null,
      status: null,
      ssn_or_itin: null,
    };
  }
}

export default User;
