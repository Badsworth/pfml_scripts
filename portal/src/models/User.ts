/* eslint sort-keys: ["error", "asc"] */
export const PhoneType = {
  cell: "Cell",
  phone: "Phone",
} as const;

export const MFAPreference = {
  optOut: "Opt Out",
  sms: "SMS",
} as const;

class User {
  auth_id: string;
  consented_to_data_sharing: boolean;
  email_address: string;
  roles: UserRole[] = [];
  user_id: string;
  mfa_phone_number: {
    int_code: string | null;
    phone_number: string | null;
    phone_type: typeof PhoneType[keyof typeof PhoneType] | null;
  } | null = null;

  mfa_delivery_preference:
    | typeof MFAPreference[keyof typeof MFAPreference]
    | null = null;

  user_leave_administrators: UserLeaveAdministrator[] = [];

  constructor(attrs: Partial<User>) {
    Object.assign(this, attrs);
  }

  get hasEmployerRole() {
    return this.roles.some(
      (userRole) => userRole.role_description === RoleDescription.employer
    );
  }

  /**
   * Determines whether user_leave_administrators has only unverified employers
   */
  get hasOnlyUnverifiedEmployers(): boolean {
    return this.verifiedEmployers.length === 0;
  }

  /**
   * Determines whether user_leave_administrators has a verifiable employer
   */
  get hasVerifiableEmployer(): boolean {
    return this.user_leave_administrators.some((employer) =>
      this.isVerifiableEmployer(employer)
    );
  }

  /**
   * Returns list of verified employers
   */
  get verifiedEmployers(): UserLeaveAdministrator[] {
    return this.user_leave_administrators.filter(
      (employer) => employer.verified === true
    );
  }

  /**
   * Determines whether an employer is verifiable (unverified and has verification data)
   */
  isVerifiableEmployer(employer: UserLeaveAdministrator): boolean {
    return !employer.verified && employer.has_verification_data;
  }

  /**
   * Determines whether an employer is registered in FINEOS
   */
  isEmployerIdRegisteredInFineos(employerId: string | null): boolean {
    return this.user_leave_administrators.some(
      (employer) =>
        employerId === employer.employer_id && employer.has_fineos_registration
    );
  }
}

export class UserRole {
  role_description: typeof RoleDescription[keyof typeof RoleDescription];
  role_id: number;

  constructor(attrs: Partial<UserRole>) {
    Object.assign(this, attrs);
  }
}

/**
 * Enums for UserRole's `role_description` field
 * @enum {string}
 */
export const RoleDescription = {
  claimant: "Claimant",
  employer: "Employer",
} as const;

export class UserLeaveAdministrator {
  employer_dba: string;
  employer_fein: string;
  employer_id: string;
  has_fineos_registration: boolean;
  has_verification_data: boolean;
  verified: boolean;

  constructor(attrs: Partial<UserLeaveAdministrator>) {
    Object.assign(this, attrs);
  }
}

export default User;
