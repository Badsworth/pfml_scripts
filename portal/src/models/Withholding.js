import BaseModel from "./BaseModel";

class Withholding extends BaseModel {
  get defaults() {
    return {
      filing_period: null,
    };
  }
}

export default Withholding;
