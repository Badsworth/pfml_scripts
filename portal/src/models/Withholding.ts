interface Withholding {
  filing_period: string;
}

export default Withholding;
