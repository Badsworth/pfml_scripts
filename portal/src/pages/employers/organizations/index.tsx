import withUser, { WithUserProps } from "../../../hoc/withUser";
import Alert from "../../../components/core/Alert";
import BackButton from "../../../components/BackButton";
import ButtonLink from "../../../components/ButtonLink";
import LeaveAdministratorRow from "../../../components/employers/LeaveAdministratorRow";
import React from "react";
import Table from "../../../components/core/Table";
import Title from "../../../components/core/Title";
import { Trans } from "react-i18next";
import routes from "../../../routes";
import { useTranslation } from "../../../locales/i18n";

export const Index = (
  props: WithUserProps & { query: { account_converted?: string } }
) => {
  const { appLogic, query } = props;
  const { t } = useTranslation();
  const { hasVerifiableEmployer, user_leave_administrators } = props.user;
  const accountConverted = query?.account_converted === "true";

  return (
    <React.Fragment>
      <BackButton
        label={t("pages.employersOrganizations.backToDashboardLabel")}
        href={appLogic.portalFlow.getNextPageRoute("BACK")}
      />
      <Title>{t("pages.employersOrganizations.title")}</Title>
      {accountConverted && (
        <Alert
          heading={t("pages.employersOrganizations.convertHeading")}
          state="success"
        >
          {t("pages.employersOrganizations.convertDescription")}
        </Alert>
      )}
      {hasVerifiableEmployer && (
        <Alert
          state="warning"
          heading={t("pages.employersOrganizations.verificationTitle")}
        >
          <p>
            <Trans
              i18nKey="pages.employersOrganizations.verificationBody"
              components={{
                "your-organizations-link": <React.Fragment />,
              }}
            />
          </p>
        </Alert>
      )}
      <p data-test="future-availability-message">
        {t("pages.employersOrganizations.nearFutureAvailability")}
      </p>

      <Table responsive className="width-full">
        <thead>
          <tr>
            <th scope="col">
              {t("pages.employersOrganizations.organizationsTableHeader")}
            </th>
            <th scope="col">
              {t("pages.employersOrganizations.einTableHeader")}
            </th>
          </tr>
        </thead>
        <tbody>
          {user_leave_administrators.length > 0 &&
            user_leave_administrators.map((leaveAdmin) => (
              <LeaveAdministratorRow
                key={leaveAdmin.employer_id}
                leaveAdmin={leaveAdmin}
              />
            ))}
          {user_leave_administrators.length === 0 && (
            <tr>
              <td>{t("shared.noneReported")}</td>
              <td />
            </tr>
          )}
        </tbody>
      </Table>
      <ButtonLink href={routes.employers.addOrganization}>
        {t("pages.employersOrganizations.addOrganizationButton")}
      </ButtonLink>
    </React.Fragment>
  );
};

export default withUser(Index);
