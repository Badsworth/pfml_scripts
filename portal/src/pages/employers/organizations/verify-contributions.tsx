import withWithholding, {
  WithWithholdingProps,
} from "../../../hoc/withWithholding";
import Button from "../../../components/core/Button";
import Details from "../../../components/core/Details";
import InputCurrency from "../../../components/core/InputCurrency";
import Lead from "../../../components/core/Lead";
import React from "react";
import Title from "../../../components/core/Title";
import { Trans } from "react-i18next";
import formatDateRange from "../../../utils/formatDateRange";
import routes from "../../../routes";
import useFormState from "../../../hooks/useFormState";
import useFunctionalInputProps from "../../../hooks/useFunctionalInputProps";
import useThrottledHandler from "../../../hooks/useThrottledHandler";
import { useTranslation } from "../../../locales/i18n";

interface VerifyContributionsProps extends WithWithholdingProps {
  query: { next?: string };
}

export const VerifyContributions = (props: VerifyContributionsProps) => {
  const { appLogic, employer, query, withholding } = props;
  const { t } = useTranslation();

  const { formState, updateFields } = useFormState({
    withholdingAmount: "",
  });

  const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value;
    const amount = value ? Number(value.replace(/,/g, "")) : 0;
    updateFields({ withholdingAmount: amount });
  };

  const handleSubmit = useThrottledHandler(async (event) => {
    event.preventDefault();

    const payload = {
      employer_id: employer.employer_id,
      withholding_amount: formState.withholdingAmount,
      withholding_quarter: withholding.filing_period,
    };

    await appLogic.employers.submitWithholding(payload, query.next);
  });

  const getFunctionalInputProps = useFunctionalInputProps({
    appErrors: appLogic.appErrors,
    formState,
    updateFields,
  });

  return (
    <form className="usa-form" onSubmit={handleSubmit} method="post">
      <Title>
        {t("pages.employersOrganizationsVerifyContributions.title")}
      </Title>
      <Lead>
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.companyNameLabel"
          tOptions={{ company: employer.employer_dba }}
        />
        <br />
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.employerIdNumberLabel"
          tOptions={{ ein: employer.employer_fein }}
        />
      </Lead>
      <Lead>
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.lead"
          components={{
            "mass-tax-connect-link": (
              <a
                href={routes.external.massTaxConnect}
                target="_blank"
                rel="noreferrer noopener"
              />
            ),
          }}
        />
      </Lead>
      <Details
        label={t(
          "pages.employersOrganizationsVerifyContributions.detailsLabel"
        )}
      >
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.detailsList"
          values={{
            date: formatDateRange(withholding.filing_period),
          }}
          components={{
            "mass-tax-connect-link": (
              <a
                href={routes.external.massTaxConnect}
                target="_blank"
                rel="noreferrer noopener"
              />
            ),
            "dor-phone-link": (
              <a href={`tel:${t("shared.departmentOfRevenuePhoneNumber")}`} />
            ),
            "recent-filing-periods-link": (
              <a
                href={routes.external.massgov.preparingToVerifyEmployer}
                target="_blank"
                rel="noreferrer noopener"
              />
            ),
            ol: <ol className="usa-list" />,
            li: <li />,
          }}
        />
      </Details>
      <InputCurrency
        {...getFunctionalInputProps("withholdingAmount")}
        onChange={handleAmountChange}
        hint={t(
          "pages.employersOrganizationsVerifyContributions.withholdingAmountHint"
        )}
        label={t(
          "pages.employersOrganizationsVerifyContributions.withholdingAmountLabel",
          {
            date: formatDateRange(withholding.filing_period),
          }
        )}
        smallLabel
      />
      <Button type="submit" loading={handleSubmit.isThrottled}>
        {t("pages.employersOrganizationsVerifyContributions.submitButton")}
      </Button>
    </form>
  );
};

export default withWithholding(VerifyContributions);
