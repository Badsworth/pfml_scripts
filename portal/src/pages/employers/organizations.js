import BackButton from "../../components/BackButton";
import PropTypes from "prop-types";
import React from "react";
import Table from "../../components/Table";
import Tag from "../../components/Tag";
import Title from "../../components/Title";
import User from "../../models/User";
import { useTranslation } from "../../locales/i18n";
import withUser from "../../hoc/withUser";

export const Organizations = ({ appLogic }) => {
  const { t } = useTranslation();
  const { user_leave_administrators } = appLogic.users.user;

  return (
    <React.Fragment>
      <BackButton />
      <Title>{t("pages.employersOrganizations.title")}</Title>
      <p>{t("pages.employersOrganizations.nearFutureAvailability")}</p>
      <Table className="width-full">
        <thead>
          <tr>
            <th scope="col">
              {t("pages.employersOrganizations.organizationsTableHeader")}
            </th>
            <th scope="col">
              {t("pages.employersOrganizations.einTableHeader")}
            </th>
          </tr>
        </thead>
        <tbody>
          {user_leave_administrators.length > 0 &&
            user_leave_administrators.map((leaveAdmin) => (
              <LeaveAdministratorRow
                key={leaveAdmin.employer_id}
                {...leaveAdmin}
              />
            ))}
          {user_leave_administrators.length === 0 && (
            <tr>
              <th scope="row">
                <span>{t("shared.noneReported")}</span>
              </th>
            </tr>
          )}
        </tbody>
      </Table>
    </React.Fragment>
  );
};

Organizations.propTypes = {
  appLogic: PropTypes.shape({
    users: PropTypes.shape({
      user: PropTypes.instanceOf(User).isRequired,
    }).isRequired,
  }).isRequired,
};

const LeaveAdministratorRow = ({ employer_dba, employer_fein, verified }) => {
  const { t } = useTranslation();
  const employerDbaClassName = !verified ? "margin-right-3" : undefined;

  return (
    <tr>
      <th scope="row">
        <span className={employerDbaClassName}>{employer_dba}</span>
        {!verified && (
          <Tag
            label={t("pages.employersOrganizations.verificationRequired")}
            state="warning"
          />
        )}
      </th>
      <td>{employer_fein}</td>
    </tr>
  );
};

LeaveAdministratorRow.propTypes = {
  employer_dba: PropTypes.string.isRequired,
  employer_fein: PropTypes.string.isRequired,
  verified: PropTypes.bool.isRequired,
};

export default withUser(Organizations);
