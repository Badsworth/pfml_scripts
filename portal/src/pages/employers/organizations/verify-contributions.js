import AppErrorInfoCollection from "../../../models/AppErrorInfoCollection";
import Button from "../../../components/Button";
import Details from "../../../components/Details";
import InputCurrency from "../../../components/InputCurrency";
import Lead from "../../../components/Lead";
import PropTypes from "prop-types";
import React from "react";
import Title from "../../../components/Title";
import { Trans } from "react-i18next";
import User from "../../../models/User";
import Withholding from "../../../models/Withholding";
import formatDateRange from "../../../utils/formatDateRange";
import routes from "../../../routes";
import useFormState from "../../../hooks/useFormState";
import useFunctionalInputProps from "../../../hooks/useFunctionalInputProps";
import useThrottledHandler from "../../../hooks/useThrottledHandler";
import { useTranslation } from "../../../locales/i18n";
import withUser from "../../../hoc/withUser";
import withWithholding from "../../../hoc/withWithholding";

export const VerifyContributions = (props) => {
  const { appLogic, query, withholding } = props;
  const {
    users: { user },
  } = appLogic;
  const { t } = useTranslation();
  const employer = user.user_leave_administrators.find((employer) => {
    return employer.employer_id === query.employer_id;
  });

  const { formState, updateFields } = useFormState({
    withholdingAmount: 0,
  });

  const handleAmountChange = (event) => {
    const value = event.target.value;
    const amount = value ? Number(value.replace(/,/g, "")) : 0;
    updateFields({ withholdingAmount: amount });
  };

  const handleSubmit = useThrottledHandler(async (event) => {
    event.preventDefault();

    const payload = {
      employer_id: query.employer_id,
      withholding_amount: formState.withholdingAmount,
      withholding_quarter: withholding.filing_period,
    };

    await appLogic.employers.submitWithholding(payload, query.next);
  });

  const getFunctionalInputProps = useFunctionalInputProps({
    appErrors: appLogic.appErrors,
    formState,
    updateFields,
  });

  return (
    <form className="usa-form" onSubmit={handleSubmit} method="post">
      <Title>
        {t("pages.employersOrganizationsVerifyContributions.title")}
      </Title>
      <Lead>
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.companyNameLabel"
          tOptions={{ company: employer.employer_dba }}
        />
        <br />
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.employerIdNumberLabel"
          tOptions={{ ein: employer.employer_fein }}
        />
      </Lead>
      <Lead>
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.lead"
          components={{
            "mass-tax-connect-link": (
              <a
                href={routes.external.massTaxConnect}
                target="_blank"
                rel="noopener"
              />
            ),
          }}
        />
      </Lead>
      <Details
        label={t(
          "pages.employersOrganizationsVerifyContributions.detailsLabel"
        )}
      >
        <Trans
          i18nKey="pages.employersOrganizationsVerifyContributions.detailsList"
          values={{
            date: formatDateRange(withholding.filing_period),
          }}
          components={{
            "mass-tax-connect-link": (
              <a
                href={routes.external.massTaxConnect}
                target="_blank"
                rel="noopener"
              />
            ),
            "dor-phone-link": (
              <a href={`tel:${t("shared.departmentOfRevenuePhoneNumber")}`} />
            ),
            "recent-filing-periods-link": (
              <a
                href={routes.external.massgov.preparingToVerifyEmployer}
                target="_blank"
                rel="noopener"
              />
            ),
            ol: <ol className="usa-list" />,
            li: <li />,
          }}
        />
      </Details>
      <InputCurrency
        {...getFunctionalInputProps("withholdingAmount")}
        onChange={handleAmountChange}
        mask="currency"
        hint={t(
          "pages.employersOrganizationsVerifyContributions.withholdingAmountHint"
        )}
        label={t(
          "pages.employersOrganizationsVerifyContributions.withholdingAmountLabel",
          {
            date: formatDateRange(withholding.filing_period),
          }
        )}
        smallLabel
      />
      <Button type="submit" loading={handleSubmit.isThrottled}>
        {t("pages.employersOrganizationsVerifyContributions.submitButton")}
      </Button>
    </form>
  );
};

VerifyContributions.propTypes = {
  appLogic: PropTypes.shape({
    appErrors: PropTypes.instanceOf(AppErrorInfoCollection),
    employers: PropTypes.shape({
      submitWithholding: PropTypes.func.isRequired,
    }),
    portalFlow: PropTypes.shape({
      goTo: PropTypes.func.isRequired,
    }).isRequired,
    users: PropTypes.shape({
      user: PropTypes.instanceOf(User),
    }).isRequired,
  }).isRequired,
  query: PropTypes.shape({
    employer_id: PropTypes.string.isRequired,
    next: PropTypes.string,
  }).isRequired,
  withholding: PropTypes.instanceOf(Withholding).isRequired,
};

export default withUser(withWithholding(VerifyContributions));
