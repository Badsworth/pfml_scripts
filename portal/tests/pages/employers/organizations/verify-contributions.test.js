import User, { UserLeaveAdministrator } from "../../../../src/models/User";
import { screen, waitFor } from "@testing-library/react";
import { VerifyContributions } from "../../../../src/pages/employers/organizations/verify-contributions";
import Withholding from "../../../../src/models/Withholding";
import { renderPage } from "../../../test-utils";
import userEvent from "@testing-library/user-event";

const query = {
  employer_id: "mock_employer_id",
  next: "/employers/applications/new-application/?absence_id=mock_absence_id",
};

const setup = (props = {}) => {
  let submitWithholdingSpy;
  const utils = renderPage(
    VerifyContributions,
    {
      addCustomSetup: (appLogic) => {
        appLogic.users.user = new User({
          consented_to_data_sharing: true,
          user_leave_administrators: [
            new UserLeaveAdministrator({
              employer_dba: "Company Name",
              employer_fein: "12-3456789",
              employer_id: "mock_employer_id",
              verified: false,
            }),
          ],
        });
        submitWithholdingSpy = jest.spyOn(
          appLogic.employers,
          "submitWithholding"
        );
      },
    },
    {
      query,
      withholding: new Withholding({
        filing_period: "2011-11-20",
      }),
      ...props,
    }
  );
  return { submitWithholdingSpy, ...utils };
};

describe("VerifyContributions", () => {
  it("renders the page", () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("submits withholding data with correct values", async () => {
    const { submitWithholdingSpy } = setup();
    userEvent.type(screen.getByRole("textbox"), "1,234.560");
    userEvent.click(screen.getByRole("button", { name: "Submit" }));

    await waitFor(() => {
      expect(submitWithholdingSpy).toHaveBeenCalledWith(
        {
          employer_id: "mock_employer_id",
          withholding_amount: 1234.56,
          withholding_quarter: "2011-11-20",
        },
        query.next
      );
    });
  });

  it("submits withholding data as 0 if input is invalid", async () => {
    const { submitWithholdingSpy } = setup();
    userEvent.type(screen.getByRole("textbox"), "invalid");
    userEvent.click(screen.getByRole("button", { name: "Submit" }));

    await waitFor(() => {
      expect(submitWithholdingSpy).toHaveBeenCalledWith(
        {
          employer_id: "mock_employer_id",
          withholding_amount: 0,
          withholding_quarter: "2011-11-20",
        },
        query.next
      );
    });
  });
});
