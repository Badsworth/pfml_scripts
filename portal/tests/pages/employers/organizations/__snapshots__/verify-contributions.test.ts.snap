// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyContributions renders the page 1`] = `
<form
  class="usa-form"
  method="post"
>
  <h1
    class="js-title margin-top-0 margin-bottom-2 font-heading-lg line-height-sans-2"
    tabindex="-1"
  >
    Verify your paid leave contributions from MassTaxConnect
  </h1>
  <p
    class="usa-intro"
  >
    <strong>
      Organization:
    </strong>
     Company Name
    <br />
    <strong>
      Employer ID number (EIN):
    </strong>
     12-3456789
  </p>
  <p
    class="usa-intro"
  >
    We need more information to verify your identity. We require every employer to verify recent 
    <a
      href="https://mtc.dor.state.ma.us/mtc/_/"
      rel="noreferrer noopener"
      target="_blank"
    >
      MassTaxConnect
    </a>
     data when creating an account. This helps protect your employees and your organization's information.
  </p>
  <details
    class="c-details"
  >
    <summary
      class="margin-bottom-2 text-primary text-underline font-ui-xs"
    >
      Where to find your paid leave contributions
    </summary>
    <div
      class="margin-bottom-2"
    >
      Log into 
      <a
        href="https://mtc.dor.state.ma.us/mtc/_/"
        rel="noreferrer noopener"
        target="_blank"
      >
        MassTaxConnect
      </a>
       or contact your payroll department to complete these steps:
      <ol
        class="usa-list"
      >
        <li>
          On the 
          <strong>
            Summary
          </strong>
           page, scroll down to the 
          <strong>
            Paid Family and Medical Leave
          </strong>
           section on the left. In the 
          <strong>
            Account
          </strong>
           portion, select 
          <strong>
            Returns
          </strong>
          .
        </li>
        <li>
          Choose 
          <strong>
            the last period
          </strong>
           for which a return has been 
          <strong>
            received
          </strong>
          . For example, if you sent your contributions for 3-31-2021, and the return has been processed and designated as ‘received’, you can use the amount from that period to verify your account. If you have not yet sent your contributions or it is still being processed, use the amount from the most recent period for which you filed that has been processed.
        </li>
        <li>
          Go into the return. Click 
          <strong>
            View or Amend Return
          </strong>
          . Then select 
          <strong>
            Next
          </strong>
           at the bottom. Look to 
          <strong>
            line 6
          </strong>
           and you will find the 
          <strong>
            Total Contributions Due
          </strong>
          .
        </li>
        <li>
          Copy the 
          <strong>
            Total Contributions Due
          </strong>
           amount for verification.
        </li>
      </ol>
      If you have any questions about your paid leave contributions, please contact the Department of Revenue at 
      <a
        href="tel:(617) 466‑3950"
      >
        <strong>
          (617) 466‑3950
        </strong>
      </a>
       from 9am-4pm ET.
    </div>
  </details>
  <div
    class="usa-form-group"
  >
    <label
      class="usa-label text-bold font-heading-xs measure-5"
      for="InputText1"
      id="InputText1_label"
    >
      Paid leave contributions from 11/20/2011
    </label>
    <span
      class="display-block line-height-sans-5 measure-5 usa-hint text-base-darkest"
      id="InputText1_hint"
    >
      Include the full amount with dollars and cents.
    </span>
    <div
      class="usa-input-group"
    >
      <div
        aria-hidden="true"
        class="usa-input-prefix"
      >
        <svg
          aria-hidden="true"
          class="usa-icon"
          focusable="false"
          role="img"
        >
          <use
            xlink:href="/img/sprite.svg#attach_money"
          />
        </svg>
      </div>
      <input
        aria-labelledby="InputText1_label InputText1_hint InputText1_error"
        class="usa-input maxw-mobile-lg padding-left-0"
        data-value-type="float"
        id="InputText1"
        inputmode="decimal"
        name="withholdingAmount"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="usa-button position-relative"
    type="submit"
  >
    Submit
  </button>
</form>
`;
